src/hotspot/share/opto/gcm.cpp

```diff
@@ -686,7 +710,6 @@ Block* PhaseCFG::insert_anti_dependences(Block* LCA, Node* load, bool verify) {
     }
 
     if (op == Op_MachProj || op == Op_Catch)   continue;
-    if (store->needs_anti_dependence_check())  continue;  // not really a store
```

Comment:
I didn't remove it, I moved it. It's what this:

```
The fix I propose for this is to test before pushing on the work list
whether a node is a store or not.
```

refers to. What we do today is that we push all use and then filter out what's not of interest. What I propose is to filter out what's not useful before it's pushed so the queue doesn't grow large with nodes that are going to be discarded anyway.

Reply:
That's what I did but the assert is where the loop starts not where the line was removed.

Reply:
Hmm, I did see that. But that is quite far away, and who knows maybe someone in the future breaks that assumption somewhere inside the loop. Therefore, it would be nice and easy to restate that assumption.